--- src/main/java/net/lax1dude/eaglercraft/v1_8/touch_gui/TouchControls.java
+++ src/main/java/net/lax1dude/eaglercraft/v1_8/touch_gui/TouchControls.java
@@ -54,6 +54,12 @@
					loc[1] *= fac;
					int size = ctrl.getSize() * fac;
					if (x >= loc[0] && y >= loc[1] && x < loc[0] + size && y < loc[1] + size) {
+						input.x = x / fac;
+						input.y = y / fac;
+						continue;
+					} else if (ctrl == EnumTouchControl.JOYSTICK) {
+						input.x = x / fac;
+						input.y = y / fac;
						continue;
					}
					EnumTouchControl[] en = EnumTouchControl._VALUES;
